eng/pipelines/templates/steps/ref-updater.yml

```diff
@@ -8,28 +8,38 @@ parameters:
 steps:
 - ${{ each repo in parameters.Repos }}:
   - pwsh: |
-      git clone https://github.com/azure/${{ repo }}
-    displayName: Clone ${{ repo }}
+      git clone https://github.com/azure/${{ repo.key }}
+      if ("${{ repo.value.branch }}")
+      { 
+        Push-Location ${{ repo.key }}
+        Write-Host "git checkout -b ${{ repo.value.branch }} origin/${{ repo.value.branch }}"
+        git checkout ${{ repo.value.branch }}
+        Pop-Location
+      }
+    displayName: Clone ${{ repo.key }}
     workingDirectory: $(System.DefaultWorkingDirectory)
   - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
   - task: Powershell@2
     inputs:
       targetType: 'filePath'
       filePath: ${{ parameters.ToolsRepoPath }}/scripts/powershell/ref-updater.ps1
       arguments: >
-        -RepoRoot $(System.DefaultWorkingDirectory)/${{ repo }}
+        -RepoRoot $(System.DefaultWorkingDirectory)/${{ repo.key }}
         -Tag ${{ parameters.Tag }}
         -ToolRepo ${{ parameters.ToolRepo }}
       pwsh: true
     displayName: Update Refs
 
   - template: ../../../common/pipelines/templates/steps/create-pull-request.yml
     parameters:
-      RepoName: ${{ repo }}
+      RepoName: ${{ repo.key }}
       PRBranchName: UpdateRepositoryRefsFor${{ parameters.ToolRepo }}
       CommitMsg: Update ${{ parameters.ToolRepo }} Repository Resource Refs in Yaml files
       PRTitle: Update ${{ parameters.ToolRepo }} Repository Resource Refs in Yaml files
       PushArgs: -f
-      WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
+      WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo.key }}
       ScriptDirectory: ${{ parameters.ToolsRepoPath }}/eng/common/scripts
-      BaseBranchName: "$(DefaultBranch)"
+      ${{ if ne(repo.value.branch, '') }}:
```

Comment:
In what places would `repo.value.branch` get set? I think it might be clearer to set the default branch value for all the repos and do away with this conditional.

Reply:
its going to come from the build tools repo. https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools/pullrequest/201